lib/facility.rb

```diff
@@ -1,14 +1,62 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicle, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_information)
+    @name = facility_information[:name]
+    @address = facility_information[:address]
+    @phone = facility_information[:phone]
     @services = []
+    @registered_vehicle = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      @registered_vehicle << vehicle
+      vehicle.car_registration_date
+      collect_fees(vehicle)
+      @registration_date = vehicle.car_registration_date
+    end
+  end
+
+  def administer_written_test(registrant)
+    if registrant.age >= 16 && @services.include?("Written Test") && registrant.permit?
+      registrant.license_data[:written] = true
+      true
+    else
+      false
+    end
+  end
+
+
+  def administer_road_test(registrant)
+    if registrant.age >= 16 && @services.include?("Road Test") && registrant.permit? && registrant.license_data[:written] == true
```

Comment:
don't need the `== true` at the end since the expression `registrant.license_data[:written]` will already return a true or false.